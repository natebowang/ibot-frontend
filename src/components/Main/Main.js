import React from "react";
import {FOOTER_HEIGHT, NAV_BAR_HEIGHT} from "../../app/config";

const Main = ({children})=>(
    <main style={mainStyle}>
        {children}
    </main>
);

const mainStyle = {
    display: 'flex',
    flexDirection: 'column',
    height: `calc( 100vh - ${FOOTER_HEIGHT} - ${NAV_BAR_HEIGHT} )`,
    // minHeight: `calc( 100vh - ${FOOTER_HEIGHT} - ${NAV_BAR_HEIGHT} )`,
};

export default Main;