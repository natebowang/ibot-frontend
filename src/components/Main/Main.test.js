import React from "react";
import {create} from 'react-test-renderer';
import Main from "./Main";

test('Main component snapshot test', () => {
    const main = create(
        <Main>test</Main>
    );
    const tree = main.toJSON();
    expect(tree).toMatchSnapshot();
});