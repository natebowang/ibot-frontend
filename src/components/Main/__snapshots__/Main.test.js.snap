// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component snapshot test 1`] = `
<main
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "minHeight": "calc( 100vh - 12.5rem - 4rem )",
    }
  }
>
  test
</main>
`;
